cpp/src/IceGrid/ServerAdapterI.h

```diff
@@ -19,37 +19,37 @@ namespace IceGrid
             const std::shared_ptr<NodeI>&,
             ServerI*,
             const std::string&,
-            const AdapterPrxPtr&,
+            AdapterPrx,
             const std::string&,
             bool);
         ~ServerAdapterI() override;
 
         void activateAsync(
-            std::function<void(const Ice::ObjectPrxPtr&)>, // TODO: pass by value!
+            std::function<void(const std::optional<Ice::ObjectPrx>&)>, // TODO: pass by value!
```

Comment:
Is this TODO still relevant?

Reply:
Yes, the callback is still taking a reference as before.